src/Abp.ZeroCore.EntityFrameworkCore/EntityHistory/EntitySnapshotManager.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Abp.Dependency;
+using Abp.Domain.Entities;
+using Abp.Domain.Repositories;
+using Abp.EntityFramework;
+using Abp.EntityFrameworkCore.Repositories;
+using Abp.Events.Bus.Entities;
+using Abp.Json;
+using Microsoft.EntityFrameworkCore;
+
+namespace Abp.EntityHistory
+{
+    public class EntitySnapshotManager : IEntitySnapshotManager, ITransientDependency
+    {
+        private readonly IRepository<EntityChange, long> _entityChangeRepository;
+
+        public EntitySnapshotManager(IRepository<EntityChange, long> entityChangeRepository)
+        {
+            _entityChangeRepository = entityChangeRepository;
+        }
+        public async Task<EntityHistorySnapshot> GetSnapshotAsync<TEntity>(int id, DateTime snapshotTime) where TEntity : class, IEntity<int>
```

Comment:
can move this into `IEntitySnapshotManager` extensions

Reply:
Do you think should i create an extensions for that? There is no extension for that now.